### PR TITLE
New serverless pattern apigw-lambda-translate

### PR DESCRIPTION
**Pull request by sudheery80**
*Issue #, if available:*
https://github.com/aws-samples/serverless-patterns/issues/965
*Description of changes:*
This pattern creates an Amazon API Gateway REST API and an AWS Lambda function that calls AWS Translate service.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apigw-lambda-translate/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## How it works
+
+This pattern deploys an Amazon API Gateway REST API and a default route integrated with an AWS Lambda function written in Python. The lambda function calls AWS Translate service using the text and language given in the request body and returns the translated text or error response if the language code is not supported.
+
+## Testing
+
+Once the application is deployed, either use a curl or call the endpoint from Postman.
+
+Example POST Request to translate text to Spanish:
+    curl -X POST "https://{api-gateway-endpoint}/prod"  -H 'Content-Type: application/json' -d '{"OriginalText":"Hello, my name is AWS","TranslateToLanguage":"es"}'
```
Comment (mavi888) [inline apigw-lambda-translate/README.md:49]: Make this text as code

File: apigw-lambda-translate/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+Response:
+```
+  "Hola, mi nombre es AWS"
+```
+
+Example POST Request to translate text to Afrikaans:
+    curl -X POST "https://{api-gateway-endpoint}/prod"  -H 'Content-Type: application/json' -d '{"OriginalText":"Hello, my name is AWS","TranslateToLanguage":"af"}'
```
Comment (mavi888) [inline apigw-lambda-translate/README.md:57]: make this text formated as code

File: apigw-lambda-translate/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Response:
+```
+  "Hallo, my naam is AWS"
+```
+
+Example POST Request with unsupported language:
+    curl -X POST "https://{api-gateway-endpoint}/prod"  -H 'Content-Type: application/json' -d '{"OriginalText":"Hello, my name is AWS","TranslateToLanguage":"ef"}'
```
Comment (mavi888) [inline apigw-lambda-translate/README.md:65]: make this text formatted as code

File: apigw-lambda-translate/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+- [Working with AWS Lambda proxy integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html)
+- [AWS Lambda - the Basics](https://docs.aws.amazon.com/whitepapers/latest/serverless-architectures-lambda/aws-lambdathe-basics.html)
+- [Lambda Function Handler](https://docs.aws.amazon.com/whitepapers/latest/serverless-architectures-lambda/the-handler.html)
+- [Function Environment Variables](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html)
+
+## Cleanup
+
+1. Delete the stack
+    ```bash
+    aws cloudformation delete-stack --stack-name STACK_NAME
```
Comment (mavi888) [inline apigw-lambda-translate/README.md:83]: you can use sam delete nowadays to remove a stack, please update the readme

File: apigw-lambda-translate/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    ```
+1. Confirm the stack has been deleted
+    ```bash
+    aws cloudformation list-stacks --query "StackSummaries[?contains(StackName,'STACK_NAME')].StackStatus"
+    ```
+
+This pattern was contributed by Sudheer Yalamanchili.
+
+----
+Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (mavi888) [inline apigw-lambda-translate/README.md:93]: update to 2023

**Comment by mavi888** [comment]
Thanks for your contribution, but before i can merge this pattern please fix all the comments in this PR.

Cheers

File: apigw-lambda-translate/README.md (hunk 1 above)
Comment (sudheery80) [reply apigw-lambda-translate/README.md:50]: updated README file

File: apigw-lambda-translate/README.md (hunk 2 above)
Comment (sudheery80) [reply apigw-lambda-translate/README.md:60]: updated README file

File: apigw-lambda-translate/README.md (hunk 3 above)
Comment (sudheery80) [reply apigw-lambda-translate/README.md:70]: updated README file

File: apigw-lambda-translate/README.md (hunk 4 above)
Comment (sudheery80) [reply apigw-lambda-translate/README.md]: updated with sam delete

File: apigw-lambda-translate/README.md (hunk 5 above)
Comment (sudheery80) [reply apigw-lambda-translate/README.md]: updated to 2023

File: apigw-lambda-translate/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    * Enter a stack name
+    * Enter the desired AWS Region
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+5. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs that were created.
+
+## How it works
+
+This pattern deploys an Amazon API Gateway REST API and a default route integrated with an AWS Lambda function written in Python. The lambda function calls AWS Translate service using the text and language given in the request body and returns the translated text or error response if the language code is not supported.
```
[the remark this answers is not in the document]
Comment (sudheery80) [reply apigw-lambda-translate/README.md]: changed to LAMBDA

File: apigw-lambda-translate/template.yaml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+Transform: 'AWS::Serverless-2016-10-31'
+Description: An Amazon REST API gateway integrated with a AWS Lambda function that calls AWS Translate service to translate given text into specified language.
+
+# Global values that are applied to all applicable resources in this template
+Globals:
+  Function:
+    CodeUri: ./src
+    Runtime: python3.9
+    MemorySize: 128
+    Timeout: 15
```
[the remark this answers is not in the document]
Comment (sudheery80) [reply apigw-lambda-translate/template.yaml]: removed timeout

File: apigw-lambda-translate/template.yaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  LambdaFunction:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      FunctionName: apigw-translate
+      Handler: index.lambda_handler
+      Policies:
+      - TranslateReadOnly
+
+  # API Gateway REST API
+  RestApi:
```
[the remark this answers is not in the document]
Comment (sudheery80) [reply apigw-lambda-translate/template.yaml]: changed the code to serverless API/Function

**Comment by sudheery80** [comment]
Addressed all PR comments.

File: apigw-lambda-translate/template.yaml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+      Events:
+        ApiEvent:
+          Type: Api
+          Properties:
+            Path: /
+            Method: post
+            RestApiId:
+              Ref: ApiGatewayApi
+
+    Outputs:
```
Comment (mavi888) [inline apigw-lambda-translate/template.yaml:33]: not properly indented and this doesnt shows the output. please fix

Comment (sudheery80) [reply apigw-lambda-translate/template.yaml]: fixed indentation and outputs. Thanks

**Comment by mavi888** [comment]
Thanks for submitting this pattern. 
It is merged and you can find it in https://serverlessland.com/patterns/apigw-lambda-translate

**Pull request merged**